### ISSUE TITLE
Objects allow trailing commas, but constructors don't

### ISSUE THREAD

**Issue by jpolitz** [brownplt/pyret-lang#906]
This is allowed:

```
{x: 5, y: 10,}
```

but this isn't:

```
[list: 1, 2, 3,]
```

It seems like this ought to be consistent.

I think the trailing commas are really useful for edits, refactoring lists of elements, and @pcardune points out that they are good for diffs.

But either way, Pyret should be consistent here.

**Comment by blerner** [comment]
Agreed.  I dislike them aesthetically, but the pragmatic benefits outweigh the visual ugly.  (It's like trolling @shriram with an unclosed parenthesis.

**Comment by shriram** [comment]
I don't like unclosed parentheses, but I'm a strong advocate of trailing
commas.

Btw, shouldn't this also then allow trailing semi-colons for tuples?

**Comment by justinpombrio** [comment]
)

On 25 October 2016 at 14:31, Shriram Krishnamurthi <notifications@github.com

**Comment by jpolitz** [comment]
Tuples already made this design decision :-)

<img width="188" alt="screen shot 2016-10-25 at 2 54 53 pm" src="https://cloud.githubusercontent.com/assets/119891/19699898/18e17acc-9ac3-11e6-8ea7-a581dc0a421b.png">

**Comment by blerner** [comment]
I assume that we *don't* want to allow `[list: , ]` though, right?  (That's the same restriction we have on object literals.)
